pkg/controller/argocd/service_account.go

```diff
@@ -132,14 +132,15 @@ func (r *ReconcileArgoCD) reconcileServiceAccountClusterPermissions(name string,
 	var sa *corev1.ServiceAccount
 	var err error
 
-	if role, err = r.reconcileClusterRole(name, rules, cr); err != nil {
+	sa, err = r.reconcileServiceAccount(name, cr)
```

Comment:
Just curious why switching the order of these two reconcile func calls?

Reply:
Now the returned `Role` can also be `nil`. So I wanted to perform those checks after the ServiceAccount is created. No specific reason other than that !!